src/components/entity/ha-entity-lock.html

```diff
@@ -0,0 +1,154 @@
+<link rel="import" href="../../../bower_components/polymer/polymer-element.html">
+<link rel="import" href="../../../bower_components/paper-icon-button/paper-icon-button.html">
+
+<dom-module id="ha-entity-lock">
+  <template>
+    <style>
+      :host {
+        white-space: nowrap;
+      }
+
+      paper-icon-button {
+        color: var(--primary-text-color);
+        transition: color .5s;
+      }
+
+      paper-icon-button[state-active] {
+        color: var(--primary-color);
+      }
+
+      paper-toggle-button {
+        cursor: pointer;
+        --paper-toggle-button-label-spacing: 0;
+        padding: 13px 5px;
+        margin: -4px -5px;
+      }
+    </style>
+
+    <template is='dom-if' if='[[!stateObj.attributes.assumed_state]]'>
+      <paper-button-group selected="">
+        <paper-icon-button icon="mdi:lock-open" on-tap="unlock" state-active$='[[!isLocked]]'>Unlock
+        </paper-icon-button>
+        <paper-icon-button icon="mdi:lock" on-tap="lock" state-active$='[[isLocked]]'>Lock</paper-icon-button>
+        <template is='dom-if' if='[[isOpenable]]'>
+          <paper-icon-button icon="mdi:door-open" on-tap="open">Open</paper-icon-button>
+        </template>
+      </paper-button-group>
+    </template>
+  </template>
+</dom-module>
+
+<script>
+  class HaEntityLock extends Polymer.Element {
+    static get is() {
+      return 'ha-entity-lock';
+    }
+
+    static get properties() {
+      return {
+        hass: Object,
+        stateObj: {
+          type: Object,
+          observer: 'stateObjObserver'
+        },
+
+        toggleChecked: {
```

Comment:
You copied this from toggle. Is it needed here?

Reply:
Its needed, since it is still a toggle state model.